✨ Add OpenAPI `external_docs` parameter to `FastAPI`

cmtoro opened the pull request:
This PR adds an external_docs parameter to the __init__ method of the FastAPI class. Currently, the external_docs parameter is not being passed when generating the OpenAPI documentation, which I believe is an oversight. By including this parameter in the FastAPI class constructor, the external_docs field will be correctly included in the root of the generated OpenAPI JSON or YAML file.

Reasoning:

The external_docs field provides valuable external documentation links and should be part of the OpenAPI metadata. Including it at the root level of the OpenAPI spec enhances the usability and completeness of the generated documentation, especially for providing references to external resources.

cmtoro committed:
Add external_docs parameter to FastAPI class init

external_docs is missing in FastAPI class. Added it to __init__ method to generate openapi spec

cmtoro committed:
Add external_docs  parameters to get_openapi method

The OpenAPI class is being instantiated without the externalDocs parameter.

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

cmtoro committed:
Update test_openapi_examples.py

pre-commit-ci[bot] committed:
🎨 [pre-commit.ci] Auto format from pre-commit.com hooks

cmtoro committed:
Update applications.py

cmtoro committed:
Update test_openapi_examples.py

YuriiMotov reviewed (changes requested):
@cmtoro, thanks for your interest and efforts!

I think that description should be simplified.
Please review my suggestions

cmtoro committed:
Update fastapi/applications.py

Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

cmtoro committed:
Update fastapi/applications.py

Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

cmtoro committed:
Update fastapi/applications.py

Co-authored-by: Motov Yurii <109919500+YuriiMotov@users.noreply.github.com>

cmtoro committed:
Update applications.py

remove https://commonmark.org/ reference from openapi_external_docs documentation

cmtoro committed:
Update main.py

cmtoro committed:
Update test_application.py

cmtoro committed:
Update test_openapi_examples.py

cmtoro committed:
Merge branch 'master' into add-openapi-external-docs-parameter

YuriiMotov reviewed (approved):
LGTM

Not sure we need this parameter, but if we want to add it, then this PR effectively does it.

@cmtoro, thanks!

cmtoro committed:
Merge branch 'fastapi:master' into add-openapi-external-docs-parameter

cmtoro committed:
Merge branch 'fastapi:master' into add-openapi-external-docs-parameter

cmtoro committed:
Merge branch 'fastapi:master' into add-openapi-external-docs-parameter

cmtoro committed:
Merge branch 'fastapi:master' into add-openapi-external-docs-parameter

cmtoro committed:
Merge branch 'fastapi:master' into add-openapi-external-docs-parameter

tiangolo reviewed (approved):
Looks good, thank you! 🚀
